core/src/main/java/org/infinispan/marshall/exts/ListExternalizer.java

```diff
@@ -27,20 +28,28 @@
 
    private static final int ARRAY_LIST = 0;
    private static final int LINKED_LIST = 1;
+   private static final int EMPTY_LIST = 2;
 
-   private final IdentityIntMap<Class<?>> numbers = new IdentityIntMap<Class<?>>(2);
+   private final IdentityIntMap<Class<?>> numbers = new IdentityIntMap<>(6);
```

Comment:
I see `IdentityIntMap` rounds up the capacity to 16, but in general you should be using `CollectionFactory.computeCapacity()` to really avoid resizing.